### PR TITLE
Support additional services for gcp provider

### PR DESCRIPTION
**Pull request by RamanaReddy0M**
- closes: #330

### CODE REVIEW

**Comment by RamanaReddy0M** [comment]
```console
✗ ./cloudlist -p gcp                             

  _______             _____     __ 
 / ___/ /__  __ _____/ / (_)__ / /_
/ /__/ / _ \/ // / _  / / (_-</ __/
\___/_/\___/\_,_/\_,_/_/_/___/\__/ 

                projectdiscovery.io

[INF] Current cloudlist version 1.0.8 (latest)
[INF] Listing assets from gcp (dev) provider
gcf-sources-4200296701-us-central1.storage.googleapis.com
pd-cloudlist-test-bucket.storage.googleapis.com
us-central1-universal-team-426216-k5.cloudfunctions.net
[INF] Found 3 Hosts for gcp (dev)
```

File: pkg/providers/gcp/bucket.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+type cloudStorageProvider struct {
+	id       string
+	storage  *storage.Service
+	projects []string
+}
+
+// GetResource returns all the storage resources in the store for a provider.
+func (d *cloudStorageProvider) GetResource(ctx context.Context) (*schema.Resources, error) {
```
Comment (dogancanbakir) [inline pkg/providers/gcp/bucket.go:18]: How about something like this to make it more readable?

```go
func (d *cloudStorageProvider) GetResource(ctx context.Context) (*schema.Resources, error) {
	list := schema.NewResources()

	buckets, err := d.getBuckets()
	if err != nil {
		return nil, err
	}

	for _, bucket := range buckets {
		resource := &schema.Resource{
			ID:       d.id,
			Provider: providerName,
			DNSName:  fmt.Sprintf("%s.storage.googleapis.com", bucket.Name),
		}
		isPublic, err := d.isBucketPublic(bucket.Name)
		if err == nil && isPublic {
			resource.Public = true
		}
		list.Append(resource)
	}

	return list, nil
}
```

Comment (dogancanbakir) [reply pkg/providers/gcp/bucket.go:18]: same for other instances

**Review by dogancanbakir — APPROVED** [review]

**Review by ehsandeep — CHANGES_REQUESTED** [review]
@RamanaReddy0M we can expose service name as part of json output as well, for example:

```diff
{
  "public": true,
  "provider": "gcp",
+ "service": "s3"
  "id": "staging",
  "dns_name": "screenshot-optimized-bucket.storage.googleapis.com"
}
```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**